services/housetables/src/main/java/com/linkedin/openhouse/housetables/repository/impl/jdbc/UserTableHtsJdbcRepository.java

```diff
@@ -33,6 +34,14 @@ Optional<UserTableRow> findByDatabaseIdIgnoreCaseAndTableIdIgnoreCase(
 
   void deleteByDatabaseIdIgnoreCaseAndTableIdIgnoreCase(String databaseId, String tableId);
 
+  @Query("SELECT DISTINCT databaseId FROM UserTableRow")
```

Comment:
Yes +1 adding index on mysql table.

Reply:
Index is good. But it should be done in another PR.

Reply:
indexes are a resource stored in mysql. 

because of this requires something like a schema evolution, but AFAIK we dont execute those on the database.

I actually think the index MUST be in place before this PR, because while we may be saving in transport bytes to/from mysql, the CPU utilization will increase on mysql when mapping/reducing the scan of database_ids

Reply:
steps do this are here:

EXPLAIN SELECT DISTINCT databaseId FROM UserTableRow;

CREATE INDEX idx_databaseId ON UserTableRow(databaseId);

EXPLAIN SELECT DISTINCT databaseId FROM UserTableRow;


then post the results in this PR

Reply:
I tested on staging, and looks like it is already searching in the primary key space. Even after I added the index, it is still using the primary key. So we don't need to add the index which can add overhead to the writes.

Before:
```
mysql> EXPLAIN SELECT DISTINCT database_id FROM user_table_row;
+----+-------------+----------------+------------+-------+---------------+---------+---------+------+------+----------+--------------------------+
| id | select_type | table          | partitions | type  | possible_keys | key     | key_len | ref  | rows | filtered | Extra                    |
+----+-------------+----------------+------------+-------+---------------+---------+---------+------+------+----------+--------------------------+
|  1 | SIMPLE      | user_table_row | NULL       | range | PRIMARY       | PRIMARY | 514     | NULL |   25 |   100.00 | Using index for group-by |
+----+-------------+----------------+------------+-------+---------------+---------+---------+------+------+----------+--------------------------+
```
After
```
mysql> EXPLAIN SELECT DISTINCT database_id FROM user_table_row;
+----+-------------+----------------+------------+-------+-------------------------+---------+---------+------+------+----------+--------------------------+
| id | select_type | table          | partitions | type  | possible_keys           | key     | key_len | ref  | rows | filtered | Extra                    |
+----+-------------+----------------+------------+-------+-------------------------+---------+---------+------+------+----------+--------------------------+
|  1 | SIMPLE      | user_table_row | NULL       | range | PRIMARY,idx_database_id | PRIMARY | 514     | NULL |   24 |   100.00 | Using index for group-by |
+----+-------------+----------------+------------+-------+-------------------------+---------+---------+------+------+----------+--------------------------+
```